src/lib/hooks/useTokenList/fetchTokenList.ts

```diff
@@ -43,31 +43,38 @@ export default async function fetchTokenList(
     urls = uriToHttp(listUrl)
   }
 
+  if (urls.length === 0) {
+    throw new Error('Unrecognized list URL protocol.')
+  }
+
   for (let i = 0; i < urls.length; i++) {
     const url = urls[i]
-    const isLast = i === urls.length - 1
     let response
     try {
       response = await fetch(url, { credentials: 'omit' })
     } catch (error) {
       const message = `failed to fetch list: ${listUrl}`
       console.debug(message, error)
-      if (isLast) throw new Error(message)
       continue
     }
 
     if (!response.ok) {
       const message = `failed to fetch list: ${listUrl}`
       console.debug(message, response.statusText)
-      if (isLast) throw new Error(message)
       continue
     }
 
-    const json = await response.json()
-    const list = skipValidation ? json : await validateTokenList(json)
-    listCache?.set(listUrl, list)
-    return list
+    try {
+      const json = await response.json()
+      const list = skipValidation ? json : await validateTokenList(json)
```

Comment:
Can `validateTokenList` function throw an error? If yes, I just wanted to point out that we swallow that error and report as `Failed to parse list response`, which can be confusing. 

I'd rather wrap just the `json` part, or update the error message.

Reply:
yes i think it can. i'll update the error message to "parse and validate"